src/stats/overall.ts

```diff
@@ -1,35 +1,22 @@
-// @flow
 import _ from 'lodash';
-import { SlippiGame } from "../SlippiGame";
-import { Frames, getSinglesOpponentIndices, iterateFramesInOrder, ConversionType } from "./common";
-
-import { OverallType } from "./common";
-
-enum JoystickRegion {
-  DZ = 0,
-  NE = 1,
-  SE = 2,
-  SW = 3,
-  NW = 4,
-  N = 5,
-  E = 6,
-  S = 7,
-  W = 8,
-};
-
-export function generateOverall(game: SlippiGame): OverallType[] {
-  const playerIndices = getSinglesOpponentIndices(game);
-
-  const inputs = generateInputs(game);
-
+import { ConversionType, PlayerIndexedType, StockType, OverallType, RatioType } from "./common";
+import { PlayerInput } from './inputs';
+
+export function generateOverallStats(
+  playerIndices: PlayerIndexedType[],
+  inputs: PlayerInput[],
+  stocks: StockType[],
+  conversions: ConversionType[],
+  playableFrameCount: number,
+): OverallType[] {
```

Comment:
Worth mentioning that this function still increase in terms of processing times as the game gets longer. It wont be as bad as before but might be worth figuring out at some point how to make `overallStats` not reprocess everything every-time as well.